npm/package.js

```diff
@@ -0,0 +1,97 @@
+var shell = require('shelljs'),
+  path = require('path'),
+  async = require('async'),
+  fs = require('fs'),
+  exists,
+  codegen,
+  codegen_path,
+  getSubfolders,
+  individual_test,
+  commandOut,
+  pwd = shell.pwd();
+
+  const args = process.argv[2],
+  PATH_TO_CODEGENS_FOLDER = path.resolve(__dirname, '../codegens');
+
+// throw JS error when any shell.js command encounters an error
+shell.config.fatal=true;
+
+// ensure that the working tree is clean before packaging
+commandOut = shell.exec('source ./npm/package/require_clean_work_tree.sh');
+if (commandOut.code !== 0) {
+    shell.exit(1);
+}
+/* commandOut = shell.exec('require_clean_work_tree "create package"');
+if (commandOut.code !== 0) {
+    console.log('Working tree is not clean, please commit/stash your changes');
+    shell.exit(1);
+} */
+
+getSubfolders = (folder) => {
+    return fs.readdirSync(folder)
+      .map((subfolder) => { return { path: path.join(folder, subfolder), name: subfolder}; })
+      .filter((obj) => { return fs.statSync(obj.path).isDirectory(); });
+};
+
+individual_test = (codegen) => {
+
+    console.log(`Creating package for ${codegen}`);
+    async.series([
+        function (next) {
+            console.log(`Running deepinstall for codegen ${codegen}`);
+            commandOut = shell.exec(`npm run deepinstall ${codegen} --color always`, {silent: true});
+            if(commandOut.code !==0) {
+                console.error(`Failed to run deepinstall on codegen ${codegen}, here\'s the error:`);
+                return next(commandOut.stderr);
+            }
+            console.log(commandOut.stdout);
+            return next();
+        },
+        function (next) {
+            console.log(`Running codegen test for codegen ${codegen}`);
+            commandOut = shell.exec(`npm run test ${codegen} --color always`, {silent: true});
+            if(commandOut.code !==0) {
+                console.error(`Failed to run codegen test on codegen ${codegen}, here\'s the error:`);
+                return next(commandOut.stderr);
+            }
+            console.log(commandOut.stdout);
+            return next();
+        },
+        function (next) {
+            console.log(`Generating zip for codegen ${codegen}`);
+            commandOut = shell.exec(`npm run zip ${codegen} --color always`, {silent: true});
+            if(commandOut.code !==0) {
+                console.error(`Failed to zip codegen ${codegen}, here\'s the error:`);
+                return next(commandOut.stderr);
+            }
+            console.log(commandOut.stdout);
+        }
+    ], (err) => {
```

Comment:
This can be simply changed to `npm run test` (As done in the package.sh script). Where npm test would handle the testing of all languages. This could will be duplicated in test.js as well. Which will be difficult to maintain in case we decide to do some changes in one of them.

Reply:
@umeshp7 Yeah, we can do that. But in that case we have to assume that the user has already run the deepinstall and now he just wants to test for all the codegens.
And even in `package.sh` we are first running deepinstall and then going for test.
![image](https://user-images.githubusercontent.com/43990819/77072825-4c498c00-6a14-11ea-9ec7-2d531f1b47f3.png)

Reply:
That's okay. That's a reasonable assumption. 
You can run tests on a new repo without doing `npm i` anyway. Here the user is us/developers I think we can safely assume that deepinstall will be run before.

Reply:
Okay, I'll do that then.